src/Datadog.Trace.Ci.Shared/CIEnvironmentValues.cs

```diff
@@ -53,17 +57,60 @@ static CIEnvironmentValues()
                 SetupBuildkiteEnvironment();
             }
 
+            // **********
+            // Remove sensitive info from repository url
+            // **********
+            if (Uri.TryCreate(Repository, UriKind.Absolute, out Uri repository))
+            {
+                if (!string.IsNullOrEmpty(repository.UserInfo))
+                {
+                    Repository = repository.GetComponents(UriComponents.Fragment | UriComponents.Query | UriComponents.Path | UriComponents.Port | UriComponents.Host | UriComponents.Scheme, UriFormat.SafeUnescaped);
+                }
+            }
+
+            // **********
+            // Expand ~ in Paths
+            // **********
+
+            SourceRoot = ExpandPath(SourceRoot);
+            WorkspacePath = ExpandPath(WorkspacePath);
+
+            // **********
+            // Clean Refs
+            // **********
+
+            var regex = new Regex(@"^refs\/heads\/(.*)|refs\/tags\/(.*)|refs\/(.*)$", RegexOptions.Compiled);
```

Comment:
The regex is invoked only a couple times, compiling it is probably more expensive than just evaluating it

Reply:
done in 3215416